host/vm-manager/0001-Support-for-HPD-in-Android.patch

```diff
@@ -0,0 +1,128 @@
++	number_of_displays = (number_of_displays != 1 && number_of_displays != 2) ? 1: number_of_displays;
++	if(number_of_displays == 2) {
++		connectors_0 = g_key_file_get_string(gkf, g->name, g->key[CONNECTORS_0], &error);
++		connectors_1 = g_key_file_get_string(gkf, g->name, g->key[CONNECTORS_1], &error);
++		if (!connectors_0 && connectors_1) {
++			connectors_0 = g_key_file_get_string(gkf, g->name, g->key[CONNECTORS_1], &error);
++			number_of_displays = 1;
++                } else if (!connectors_0 && !connectors_1) {
++			number_of_displays = 1;
++		}
++	} else if(number_of_displays == 1) {
++                connectors_0 = g_key_file_get_string(gkf, g->name, g->key[CONNECTORS_0], &error);
++		if (!connectors_0) {
++			connectors_0 = g_key_file_get_string(gkf, g->name, g->key[CONNECTORS_1], &error);
++		}
++	}
++
+ 	if (error) {
+ 		if (error->code == G_KEY_FILE_ERROR_KEY_NOT_FOUND) {
+-			cx = snprintf(*p, *size, " -display gtk,gl=on");
++			if (number_of_displays == 2) {
++				if (((strstr(connectors_0, hdmi) && strlen(connectors_0) == 6) || (strstr(connectors_0, dp) && strlen(connectors_0) == 4))
++						&& (connectors_1 && ((strstr(connectors_1, hdmi) && strlen(connectors_1) == 6) || (strstr(connectors_1, dp) && strlen(connectors_1) == 4)))) {
++					cx = snprintf(*p, *size, " -display gtk,gl=on,show-fps=on,connectors.0=%s,connectors.1=%s", connectors_0, connectors_1);
++				} else if (((strstr(connectors_0, hdmi) && strlen(connectors_0) == 6) || (strstr(connectors_0, dp) && strlen(connectors_0) == 4))) {
++					cx = snprintf(*p, *size, " -display gtk,gl=on,show-fps=on,connectors.0=%s", connectors_0);
++				} else
++					cx = snprintf(*p, *size, " -display gtk,gl=on");
++			}
++			else if (number_of_displays == 1 && connectors_0 && ((strstr(connectors_0, hdmi) && strlen(connectors_0) == 6) || (strstr(connectors_0, dp) && strlen(connectors_0) == 4)))
++			{
++				cx = snprintf(*p, *size, " -display gtk,gl=on,show-fps=on,connectors.0=%s", connectors_0);
++			}
++			else {
++				cx = snprintf(*p, *size, " -display gtk,gl=on");
++			}
+ 		} else {
+ 			fprintf(stderr, "Invalid monitor id!\n");
+ 			return -1;
+@@ -557,11 +598,11 @@ static int setup_sriov(GKeyFile *gkf, char **p, size_t *size)
+ 		return -1;
+ 	vf = set_available_vf();
+ 
+-	cx = snprintf(*p, *size, " -device virtio-vga,max_outputs=1,blob=true"
+-				 " -device vfio-pci,host=0000:00:02.%d"
+-				 " -object memory-backend-memfd,hugetlb=on,id=mem_sriov,size=%dM"
+-				 " -machine memory-backend=mem_sriov",
+-				 vf, mem);
++	cx = snprintf(*p, *size, " -device virtio-vga,max_outputs=%d,blob=true"
++                               " -device vfio-pci,host=0000:00:02.%d"
++                               " -object memory-backend-memfd,hugetlb=on,id=mem_sriov,size=%dM"
++                               " -machine memory-backend=mem_sriov",
++                               number_of_displays !=0 ?number_of_displays:1, vf, mem);
```

Comment:
The number of displays should be either 1 or 2.
Throw and error if it is otherwise